zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java

```diff
@@ -147,7 +156,11 @@ public void acceptSpans(
   }
 
   void record(List<Span> sampled, Callback<Void> callback) {
-    storage.spanConsumer().accept(sampled).enqueue(callback);
+    if(blockOnStorage) {
```

Comment:
I would prefer the logic be handled here without changing the Call api. What if you just did spanConsumer.execute() instead in a try/finally where after or on exception you call the callback

Reply:
Ok, I understand. But will not be able to lock on the HttpCall semaphore, because its is not available  for this use.

Some other contention mechanism must be develop here when storage begins to drop spans

Reply:
@malonso1976, @adriancole, two things here:
1. The code here mimics the code in #2169, meaning that if both get merged a conflict will have to be resolved.  The obvious resolution being to make multiple if-conditions.  I would recommend that, instead of having any if-statements, we make use of some composition and wrap the `spanConsumer()` in some abstractions that do the appropriate things based on the settings.  
2. I think what I've been describing in #2166 can actually help resolve the current complication of what to do with the Semaphore while also giving Kafka a hook to indicate it needs to slow down.  Basically, the idea would be to us an ExecutorService with a custom RejectedExecutionHandler that would call any specific method to say slow down/retry.  I'm not familiar with Kafka so I can't say what that'd look like.  But it's an idea.
    - Further, it looks like the concurrency limiter in #2169 already has an ExecutorService internally.  If that were made a little more accessible to external configuration then we could directly share it for the purposes of pushing back on Kafka and making use of Netflix's limiter
    - It would also give me a place to put a non-zero length queue for my issue (spike load in the HTTP Collector) but could probably default to 0 for most other purposes.

I may be a bit off-base with this comment.  Especially since I'm not entirely sure if there is intent to merge both this and the Netflix limiter eventually or just one of them.  I figured I'd at least point out the conflict at the very least and hope the ExecutorService could help out.

@malonso1976, if you don't mind, I could checkout (or fork) this branch and make the change so you could see what it would look like.  As I have it implemented (locally) it already separates bulk indexing requests from GET requests so there are no load conflicts there.  Which you mentioned as a problem and I've seen myself as well.